File: src/ras/ras_timing_simple_cpu/RASTimingSimpleCPU.py
from m5.objects.TimingSimpleCPU import TimingSimpleCPU
from m5.params import *


class RASTimingSimpleCPU(TimingSimpleCPU):
    type = "RASTimingSimpleCPU"
    cxx_header = "ras/ras_timing_simple_cpu/ras_timing_simple_cpu.hh"
    cxx_class = "gem5::X86ISA::RASTimingSimpleCPU"

    shadow_stack_size = Param.Int(4096, "Shadow stack size in bytes") # Preferably a size that is a multiple of a memory page (4096 byte)

File: src/ras/ras_timing_simple_cpu/ras_timing_simple_cpu.cc
#include "ras/ras_timing_simple_cpu/ras_timing_simple_cpu.hh"

namespace gem5
{

namespace X86ISA
{

RASTimingSimpleCPU::RASTimingSimpleCPU(
    const RASTimingSimpleCPUParams& params
):
    TimingSimpleCPU(params)
{}


void RASTimingSimpleCPU::RASContainer::pushRAS(Addr returnAddress)
{
    // TODO
}

Addr RASTimingSimpleCPU::RASContainer::popRAS()
{
    // TODO
    return Addr();
}

void RASTimingSimpleCPU::RASContainer::setRAS(Addr returnAddress)
{
    // TODO
    // pass
}

} // namespace X86ISA

} // namespace gem5
